### PR TITLE
feat(lab-4341): check file type before importing

### PR DESCRIPTION
**Pull request by florianlega**

### CODE REVIEW

File: src/kili/services/asset_import/constants.py [hunk 1]
```diff
@@ -10,3 +10,29 @@
...
+            ".jp2",
+            ".ntf",
+            ".nitf",
+        }
+    ),
+    "LLM_RLHF": frozenset({".json"}),
+    "PDF": frozenset({".pdf"}),
+    "TEXT": frozenset({".txt", ".csv"}),
+    "VIDEO": frozenset({".mp4", ".mkv", ".3gp", ".avi", ".m4v", ".mov", ".webm"}),
+    "VIDEO_LEGACY": frozenset({".mp4", ".mkv", ".3gp", ".avi", ".m4v", ".mov", ".webm"}),
```
Comment (FannyGaudin) [inline src/kili/services/asset_import/constants.py]: What is this type VIDEO_LEGACY needed for ?

**Review by FannyGaudin — APPROVED** [review]

File: src/kili/services/asset_import/constants.py (hunk 1 above)
Comment (florianlega) [reply src/kili/services/asset_import/constants.py]: It's indeed not used anymore by the backend, i'll remove it

File: src/kili/services/asset_import/__init__.py [hunk 2]
```diff
@@ -35,6 +35,8 @@
     "LLM_RLHF": LLMDataImporter,
 }
 
+check_file_extensions = BaseAbstractAssetImporter.check_file_extensions
```
Comment (aurelienlombard) [inline src/kili/services/asset_import/__init__.py]: [SUGGESTION] Unused module-level alias

`check_file_extensions` is assigned here but never referenced — the actual call on line 68 uses `asset_importer.check_file_extensions(...)` (the instance/static method directly). Consider removing this alias unless you intend it as a convenience export for other modules or external callers.

File: src/kili/services/asset_import/base.py [hunk 3]
```diff
@@ -524,6 +526,43 @@ def check_asset_contents(assets: list[AssetLike]) -> None:
...
+
+        for asset in assets:
+            content = asset.get("content")
+            if isinstance(content, str):
+                ext = get_ext(content)
+                if ext and ext not in allowed:
+                    raise ImportValidationError(
+                        f"File extension '{ext}' is not allowed for {input_type} projects."
+                        f" Allowed extensions: {', '.join(sorted(allowed))}"
+                    )
```
Comment (aurelienlombard) [inline src/kili/services/asset_import/base.py:553]: [SUGGESTION] Include asset identifier in error message

When importing many assets, this error doesn't tell the user *which* asset failed. Consider including the external_id to make debugging easier:

```python
raise ImportValidationError(
    f"File extension '{ext}' is not allowed for {input_type} projects"
    f" (asset external_id='{asset.get('external_id', 'unknown')}')."
    f" Allowed extensions: {', '.join(sorted(allowed))}"
)
```

Same applies to the multi-layer error below (line 561-564).

File: src/kili/services/asset_import/constants.py [hunk 4]
```diff
@@ -10,3 +10,28 @@
...
+            ".webp",
+            ".ico",
+            ".tif",
+            ".tiff",
+            ".jp2",
+            ".ntf",
+            ".nitf",
+        }
+    ),
+    "LLM_RLHF": frozenset({".json"}),
```
Comment (aurelienlombard) [inline src/kili/services/asset_import/constants.py:33]: [QUESTION] TEXT extensions completeness

Only `.txt` and `.csv` are allowed for TEXT projects. Does this match the frontend's `fileExtensionsState` exactly? Other text-like formats (`.html`, `.md`, `.xml`) might be relevant depending on how TEXT projects are used. If the frontend also restricts to just these two, this is fine.

File: tests/unit/services/asset_import/test_import_common.py [hunk 5]
```diff
@@ -24,7 +24,8 @@ def test_cannot_upload_an_image_to_video_project(self, *_):
         url = "https://storage.googleapis.com/label-public-staging/car/car_1.jpg"
         path_image = self.downloader(url)
         assets = [{"content": path_image, "external_id": "image"}]
-        with pytest.raises(MimeTypeError):
+        # Extension check runs before MIME type check, so ImportValidationError is raised first
+        with pytest.raises(ImportValidationError):
```
Comment (aurelienlombard) [inline tests/unit/services/asset_import/test_import_common.py:28]: [QUESTION] Behavioral change: different exception type

Previously, uploading a `.jpg` to a VIDEO project raised `MimeTypeError` (from the MIME type check). Now it raises `ImportValidationError` (extension check runs first). Both exception classes inherit directly from `Exception` — they aren't related by inheritance. Any caller catching `MimeTypeError` specifically would miss this new error.

Since these are internal exceptions (not public SDK API), this is likely fine, but worth confirming no downstream code relies on catching `MimeTypeError` for this scenario.

File: tests/unit/services/asset_import/test_import_common.py [hunk 6]
```diff
@@ -84,3 +85,184 @@ def test_import_assets_verify(self, mocked_verify_batch_imported, *_):
         mocked_verify_batch_imported.assert_not_called()
         import_assets(self.kili, ProjectId("project_id"), assets, verify=True)
         mocked_verify_batch_imported.assert_called_once()
+
+
+@patch("kili.utils.bucket.generate_unique_id", mocked_unique_id)
+@patch("kili.utils.bucket.request_signed_urls", mocked_request_signed_urls)
```
Comment (aurelienlombard) [inline tests/unit/services/asset_import/test_import_common.py:91]: [PRAISE] Thorough test coverage

All input types covered with both rejection and acceptance cases, plus edge cases like extensionless URLs, raw text content, and multi-layer content validation. Well structured.

**Review by aurelienlombard — COMMENTED** [review]
## Auto Review Summary

**Verdict:** Ready to merge

| Severity | Count |
|----------|-------|
| Blocker | 0 |
| Required | 0 |
| Suggestion | 2 |
| Question | 2 |
| Praise | 1 |

Clean implementation of file extension validation in the import service. The validation logic is well-placed, correctly handles URLs (stripping query params), local paths, multi-layer content, and gracefully skips extensionless content. Tests are thorough.

**Ticket context (LAB-4341):** The ticket asks for SDK-side extension validation before calling the import service, matching the frontend behavior. The "Acceptance criteria" section is empty — consider filling it in for traceability.

File: src/kili/services/asset_import/constants.py (hunk 4 above)
Comment (aurelienlombard) [reply src/kili/services/asset_import/constants.py:33]: Not relevant i think, as for html we use rich text and use for this the jsonContent

File: tests/unit/services/asset_import/test_import_common.py (hunk 5 above)
Comment (aurelienlombard) [reply tests/unit/services/asset_import/test_import_common.py:28]: I  don't think people were using this before

**Review by aurelienlombard — APPROVED** [review]
